src/search_common.js

```diff
@@ -66,65 +65,42 @@ var bookmarks = (function(){
 	return b;
 })();
 
-var bookmarksToSuggestions = function(b, s){
-	var m = parseInt(localStorage["maxcount"]);
-	var i = 0;
-	while(s.length < m && i < b.length){
+var bookmarksToSuggestions = async function(b, s){
+	let options = await chrome.storage.sync.get(["maxcount"]);
+	var m = parseInt(options["maxcount"]);
+	for(let i = 0; s.length < m && i < b.length; i++){
 		var v = b[i];
+		if(!v.url || jsMatch.test(v.url)){
+			// javascript: bookmarklets are no longer supported,
+			// because modern websites have broken them with
+			// Content-Security-Policy: script-src
+			// https://issues.chromium.org/issues/40077444
+			continue;
+		}
```

Comment:
This method of skipping results causes a logic error. If you look further down in `searchInput`:
https://github.com/alvinhochun/chrome-bookmark-search/blob/a63793a59cbb2aab62dcbe56f861ba46bb7848cf/src/search_common.js#L126-L174
The code expects the first element (index 0) of `b` (`results`) and `s` to refer to the same bookmark. If the first result matches the condition and is skipped, then that logic produces the wrong first result.

Reply:
Ah… that is a subtle bug. Good catch!

9839f6f17ce572a0806cfda89659931a40d488dc fixes this by tagging `s` with each associated `bookmark`, instead of trying to keep `b` and `s` in sync. I considered making `s` a list of tuples, but I think annotating and then stripping out the `.bookmark` is more obvious.